codec-http2/src/main/java/io/netty/handler/codec/http2/ReadOnlyHttp2Headers.java

```diff
@@ -0,0 +1,824 @@
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers setDouble(CharSequence name, double value) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers setTimeMillis(CharSequence name, long value) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers set(Headers<? extends CharSequence, ? extends CharSequence, ?> headers) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers setAll(Headers<? extends CharSequence, ? extends CharSequence, ?> headers) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public boolean remove(CharSequence name) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers clear() {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Iterator<Map.Entry<CharSequence, CharSequence>> iterator() {
+        return new ReadOnlyIterator();
+    }
+
+    @Override
+    public Http2Headers method(CharSequence value) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers scheme(CharSequence value) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers authority(CharSequence value) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers path(CharSequence value) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public Http2Headers status(CharSequence value) {
+        throw new UnsupportedOperationException("read only");
+    }
+
+    @Override
+    public CharSequence method() {
+        return get(PseudoHeaderName.METHOD.value());
+    }
+
+    @Override
+    public CharSequence scheme() {
+        return get(PseudoHeaderName.SCHEME.value());
+    }
+
+    @Override
+    public CharSequence authority() {
+        return get(PseudoHeaderName.AUTHORITY.value());
+    }
+
+    @Override
+    public CharSequence path() {
+        return get(PseudoHeaderName.PATH.value());
+    }
+
+    @Override
+    public CharSequence status() {
+        return get(PseudoHeaderName.STATUS.value());
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder builder = new StringBuilder(getClass().getSimpleName()).append('[');
+        String separator = "";
+        for (Map.Entry<CharSequence, CharSequence> entry : this) {
+            builder.append(separator);
+            builder.append(entry.getKey()).append(": ").append(entry.getValue());
+            separator = ", ";
+        }
+        return builder.append(']').toString();
+    }
+
+    private final class ReadOnlyIterator implements Map.Entry<CharSequence, CharSequence>,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) io.netty.handler.codec.http2.ReadOnlyHttp2Headers$ReadOnlyIterator is both an Iterator and a Map.Entry [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3APZ_DONT_REUSE_ENTRY_OBJECTS_IN_ITERATORS)